cmd/tailscale/cli/file.go

```diff
@@ -413,52 +415,70 @@ func runFileGet(ctx context.Context, args []string) error {
 	if fi, err := os.Stat(dir); err != nil || !fi.IsDir() {
 		return fmt.Errorf("%q is not a directory", dir)
 	}
-
-	var wfs []apitype.WaitingFile
-	var err error
-	for {
-		wfs, err = tailscale.WaitingFiles(ctx)
-		if err != nil {
-			return fmt.Errorf("getting WaitingFiles: %w", err)
+	for true {  // Infinite loop
+		var wfs []apitype.WaitingFile
+		var err error
+		var errs []error
+		for {
+			wfs, err = tailscale.WaitingFiles(ctx)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("getting WaitingFiles: %w", err))
+			}
+			if len(wfs) != 0 || !(getArgs.wait || getArgs.loop) {
+				break
+			}
+			if getArgs.verbose {
+				printf("waiting for file...")
+			}
+			if err := waitForFile(ctx); err != nil {
+				errs = append(errs, err)
+			}
 		}
-		if len(wfs) != 0 || !getArgs.wait {
-			break
+
+		deleted := 0
+		for _, wf := range wfs {
+			writtenFile, size, err := receiveFile(ctx, wf, dir)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			if getArgs.verbose {
+				printf("wrote %v as %v (%d bytes)\n", wf.Name, writtenFile, size)
+			}
+			if err = tailscale.DeleteWaitingFile(ctx, wf.Name); err != nil {
+				errs = append(errs, fmt.Errorf("deleting %q from inbox: %v", wf.Name, err))
+				continue
+			}
+			deleted++
 		}
 		if getArgs.verbose {
-			printf("waiting for file...")
-		}
-		if err := waitForFile(ctx); err != nil {
-			return err
+			printf("moved %d/%d files\n", deleted, len(wfs))
 		}
-	}
-
-	var errs []error
-	deleted := 0
-	for _, wf := range wfs {
-		writtenFile, size, err := receiveFile(ctx, wf, dir)
-		if err != nil {
-			errs = append(errs, err)
-			continue
+		if !getArgs.loop {
+			if len(errs) == 0 {
+				return nil
+			}
+			for _, err := range errs[:len(errs)-1] {
+				outln(err)
+			}
+			return errs[len(errs)-1]
 		}
-		if getArgs.verbose {
-			printf("wrote %v as %v (%d bytes)\n", wf.Name, writtenFile, size)
+		// deamon mode
+		for _, err := range errs {
+			outln(err)
 		}
-		if err = tailscale.DeleteWaitingFile(ctx, wf.Name); err != nil {
-			errs = append(errs, fmt.Errorf("deleting %q from inbox: %v", wf.Name, err))
-			continue
+		if len(errs) > 0 {
+			// It's possible whatever caused the error(s) (e.g. conflicting target file,
+			// full disk, unwritable target directory) will re-occur if we try again so
+			// let's back off and not busy loop on error.
+			//
+			// However, let's not wait too long so other files may get received if
+			// there's just one problematic file.
+			time.Sleep(5 * time.Second)
 		}
-		deleted++
-	}
-	if getArgs.verbose {
-		printf("moved %d/%d files\n", deleted, len(wfs))
-	}
-	if len(errs) == 0 {
-		return nil
-	}
-	for _, err := range errs[:len(errs)-1] {
-		outln(err)
 	}
-	return errs[len(errs)-1]
+	// unreachable
```

Comment:
then you don't need this comment or the return line.

See https://go.dev/ref/spec#Terminating_statements

Reply:
ooooh, go's spec has a list of recognized idioms.  unreachable is... sorta turing complete, so i figured it was a "compiler not smart enough to figure it out" thing.  thanks for the pointer.